### PR TITLE
Implementation for CurvePoint: endpoints + logs + constraints

### PR DESCRIPTION
**Pull request by ALEXD2758**
Implementation of endpoints in CurvePoint Controller
Implementation CurvePointService
Modification methods CurvePointRepository
Implementation logging with LOG4J for BidListController and CurvePointController
Rectification method "validate" in BidListController and CurvePointController
Rectification of JavaDoc for all controllers (missing from previous version)
Implementation definitive comments for CurvePointController
Implementation Hibernate constraints in BidListModel and CurvePointModel

### CODE REVIEW

File: src/main/java/com/alex/poseidon/controllers/BidListController.java [hunk 1]
```diff
@@ -34,6 +38,7 @@
     @RequestMapping("/bidList/list")
     public String home(Model model) {
         model.addAttribute("bidList", bidListService.getAllBids());
+        logger.info("bidList/list : OK");
```
Comment (akath19) [inline src/main/java/com/alex/poseidon/controllers/BidListController.java]: This is missing a GET at the beginning

File: src/main/java/com/alex/poseidon/controllers/CurveController.java [hunk 2]
```diff
@@ -1,124 +1,170 @@
...
      * with attribute
      */
     @Override
     @RequestMapping("/curvePoint/list")
-    public String home(Model model)
-    {
-        // TODO: find all Curve Point, add to model
+    public String home(Model model) {
+        model.addAttribute("curvePoint", curvePointService.getAllCurvePoints());
+        logger.info("curvePoint/list : OK");
```
Comment (akath19) [inline src/main/java/com/alex/poseidon/controllers/CurveController.java]: This is missing a GET at the beginning

File: src/main/java/com/alex/poseidon/models/BidListModel.java [hunk 3]
```diff
@@ -27,34 +31,46 @@
     private double bid;
     @Column(name = "ask")
     private double ask;
+    @Size(max=125)
     @Column(name = "benchmark")
     private String benchmark;
     @Column(name = "bid_list_date")
     private Timestamp bidListDate;
```
Comment (akath19) [inline src/main/java/com/alex/poseidon/models/BidListModel.java:39]: What happens in the bid list is before now?

**Review by akath19 — CHANGES_REQUESTED** [review]
Please address comments

File: src/main/java/com/alex/poseidon/models/BidListModel.java (hunk 3 above)
Comment (ALEXD2758) [reply src/main/java/com/alex/poseidon/models/BidListModel.java:39]: Yes and no:
Yes: I forgot the constraints and logger in Bid list in story_1
No: I didn't pull the right version before, therefore my model classes changed (the name of model classes changed and references to it)

File: src/main/java/com/alex/poseidon/controllers/CurveController.java (hunk 2 above)
Comment (ALEXD2758) [reply src/main/java/com/alex/poseidon/controllers/CurveController.java]: Done

File: src/main/java/com/alex/poseidon/controllers/BidListController.java (hunk 1 above)
Comment (ALEXD2758) [reply src/main/java/com/alex/poseidon/controllers/BidListController.java]: Done

File: src/main/java/com/alex/poseidon/models/BidListModel.java (hunk 3 above)
Comment (akath19) [reply src/main/java/com/alex/poseidon/models/BidListModel.java:39]: Hmmmm....that doesn't solve my question, should this work if the bid_list_date has a date before now?

Comment (ALEXD2758) [reply src/main/java/com/alex/poseidon/models/BidListModel.java:39]: I had misunderstood your question due to line 34 having a green focus. Sorry.
I have added constraint "@FutureOrPresent" for 3 fields which I judge necessary to have a check on the date (except if revisionDate is required to be filled automatically, not by the client).

**Review by akath19 — APPROVED** [review]
LGTM

**Pull request merged**